rust/libnewsboat/src/utils.rs

```diff
@@ -360,6 +360,32 @@ pub fn get_command_output(cmd: &str) -> String {
         .unwrap_or_else(|_| String::from(""))
 }
 
+/// Extract the `filter` portion line
+/// ```
+/// use libnewsboat::utils::extract_filter;
+/// let mut filter = String::new();
+/// let mut url = String::new();
+/// extract_filter("filter:~/bin/script.sh:https://newsboat.org", &mut filter, &mut url);
+/// assert_eq!(filter, "~/bin/script.sh");
+/// assert_eq!(url, "https://newsboat.org");
+/// ```
+pub fn extract_filter(line: &str, filter: &mut String, url: &mut String) {
+    let pos = line.find(':').map(|p| p + 1);
+    let pos1 = pos.and_then(|p| line[p..].find(':').map(|p1| p1 + p));
+    println!("{}", line);
+    println!("{:?} {:?}", pos, pos1);
+    match (pos, pos1) {
+        (Some(pos), Some(pos1)) => {
+            *filter = String::from(&line[pos..pos1]);
+            *url = String::from(&line[pos1 + 1..]);
```

Comment:
This will panic if the `line` is something with an empty URL like `filter:foo:`. Should we handle that edge case?

Reply:
Actually, I'd pull this out into separate issue, because #758 takes long enough as it is—let's not add to it.

Reply:
Nevermind. I found a much simpler solution with `splitn` which works quite nice.

Reply:
I'd like to at least write a test for it once #758 is merged, so I filed #764.